### PR TITLE
Docs: Visual fixes to the site header

### PR DESCRIPTION
**Pull request by ayeshakmaz**
#### What changed?

Fix a few bugs in the header of our docs site

#### Why?

Ensure site settings dropdown always aligns properly, fix search icon in RTL, and Algolia popup in dark mode

[Tracker](https://coda.io/d/Gestalt-Team-Folder_d2LeXkQ1kVX/Gestalt-site-update-bug-tracker_sugaw#Bugs-August-18-Sprint-17_tuH4m/r4)

Before
Screen Shot 2022-09-09 at 3 52 51 PM

After
Screen Shot 2022-09-09 at 3 52 40 PM

### CODE REVIEW

File: docs/docs-components/DocSearch.js [hunk 1]
```diff
@@ -17,8 +17,21 @@ function SearchBox({ popoverZIndex }: {| popoverZIndex?: CompositeZIndex |}) {
             placeholder="Search Gestalt"
             aria-label="Search Gestalt docs"
           />
-          <Box position="absolute" top left marginTop={3} marginStart={4}>
-            <Icon icon="search" accessibilityLabel="" size="16" color="default" />
+          <Box
```
Comment (ayeshakmaz) [inline docs/docs-components/DocSearch.js:20]: Took this directly from our SearchField cmp

File: docs/docs-components/Header.js [hunk 2]
```diff
@@ -240,7 +245,7 @@ function Header() {
         </Box>
 
         <DocSearch
-          anchorRef={anchorRef}
+          anchorRef={searchAnchorRef}
```
Comment (ayeshakmaz) [inline docs/docs-components/Header.js:248]: This was the cause of the jumping Dropdown, they were sharing one anchor ref

File: packages/gestalt/src/Divider.css [hunk 3]
```diff
@@ -1,6 +1,6 @@
 .divider {
   composes: block from "./Layout.css";
-  composes: borderTop from "./Borders.css";
+  composes: borderTop solid from "./Borders.css";
```
Comment (ayeshakmaz) [inline packages/gestalt/src/Divider.css:3]: This was creating a weird artifact in dark mode for all Dividers

File: docs/docs-components/DocSearch.js (hunk 1 above)
Comment (dangerismycat) [reply docs/docs-components/DocSearch.js:20]: Can you leave a comment indicating that? So we'll know to update this if we ever need to update the SearchField markup

File: docs/docs-components/Header.js (hunk 2 above)
Comment (dangerismycat) [reply docs/docs-components/Header.js:248]: Good find!

File: docs/docs.css [hunk 4]
```diff
@@ -304,10 +304,10 @@ html[dir="rtl"] .Markdown ul + pre {
 }
 
 .algolia-autocomplete .ds-dropdown-menu {
-  background: transparent;
+  background: var(--color-background-elevation-floating);
   border: none;
   border-radius: 16px;
-  box-shadow: 0 0 8px var(--g-colorTransparentGray100);
+  box-shadow: var(--elevation-floating);
```
Comment (dangerismycat) [inline docs/docs.css:310]: ❤️

**Review by dangerismycat — APPROVED** [review]
Do you mind adding before/after screenshots to document for Future Us?

**Pull request merged**